### PR TITLE
Replace patterns with explicit class names in CondFormats `classes_def.xml`

### LINKED ISSUE #978: Replace patterns with explicit class names in CondFormats `classes_def.xml`

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR is part of series to replace the last remaining uses class pattern elements in `classes_def.xml` with explicit class names.

Resolves https://github.com/cms-sw/framework-team/issues/978

#### PR validation:

Code compiles, and the generated `.rootmap` files are identical.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
type -changes-dataformats

**Comment by makortel** [comment]
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/39803

File: CondFormats/SiPixelObjects/src/classes_def.xml [hunk 1]
```diff
@@ -77,8 +77,14 @@
   <class name="SiPixelQualityProbabilities::probabilityVec"/>
   <class name="std::map<unsigned int,SiPixelQualityProbabilities::probabilityVec>"/>
 
-  <class pattern="PixelDCSObject<*>"/> 
-  <class pattern="PixelDCSObject<*>::Item"/> 
-  <class pattern="PixelDCSObject<*>::Type"/>
```
Comment (perrotta) [inline CondFormats/SiPixelObjects/src/classes_def.xml:82]: @makortel I notice that this class pattern (with `::Type`) is not explicitated: is it intended?

Comment (makortel) [reply CondFormats/SiPixelObjects/src/classes_def.xml:82]: Yes, this is intentional. The `::Type` is an alias for the template argument itself
https://github.com/cms-sw/cmssw/blob/6262bcf9627f03ad2b1c72f28ff2392b6ae6be9c/CondFormats/SiPixelObjects/interface/PixelDCSObject.h#L21-L23
and I opted to include `CaenChannel` explicitly, whereas `bool` and `float` are handled implicitly by ROOT.

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**